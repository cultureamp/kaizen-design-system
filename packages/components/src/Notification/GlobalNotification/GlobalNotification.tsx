import React from "react"
import { NotificationType } from "~components/Notification/type"
import GenericNotification from "../subcomponents/GenericNotification"

export type GlobalNotificationProps = {
  type: NotificationType
  children: React.ReactNode
  onHide?: () => void
  persistent?: boolean
}

export const GlobalNotification = ({
  persistent = false,
  ...props
}: GlobalNotificationProps): JSX.Element => (
  <GenericNotification persistent={persistent} style="global" {...props} />
)

GlobalNotification.displayName = "GlobalNotification"
