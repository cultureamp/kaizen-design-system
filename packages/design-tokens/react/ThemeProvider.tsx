import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
