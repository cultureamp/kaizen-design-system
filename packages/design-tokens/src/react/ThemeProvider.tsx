import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
