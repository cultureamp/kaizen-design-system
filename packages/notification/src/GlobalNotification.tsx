import React from "react"
import GenericNotification, {
  NotificationType,
} from "./components/GenericNotification"

export type GlobalNotificationProps = {
  type: NotificationType
  children: React.ReactNode
  onHide?: () => void
  automationId?: string
  persistent?: boolean
}

export const GlobalNotification = (
  props: GlobalNotificationProps
): JSX.Element => <GenericNotification style="global" {...props} />
GlobalNotification.defaultProps = {
  persistent: false,
}

GlobalNotification.displayName = "GlobalNotification"
