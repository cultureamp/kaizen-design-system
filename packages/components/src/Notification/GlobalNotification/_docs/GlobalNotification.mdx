import { Canvas, Controls, Meta, Description} from "@storybook/blocks"
import { Installation, KaioNotification, ResourceLinks } from "~storybook/components"
import * as GlobalNotificationExamples from "./GlobalNotification.stories"

<Meta title="KAIO-staging/Notifications/Global Notification" />

# GlobalNotification

<ResourceLinks
  sourceCode="https://github.com/cultureamp/kaizen-design-system/tree/main/packages/components/src/Notification/GlobalNotification"
  figma="https://www.figma.com/file/eZKEE5kXbEMY3lx84oz8iN/%F0%9F%92%9C-UI-Kit%3A-Heart?type=design&node-id=1929%3A21283&mode=design&t=HmSOpdHdka19ZmTO-1"
  designGuidelines="https://cultureamp.atlassian.net/wiki/spaces/DesignSystem/pages/3082060757/Global+Notification"
  className="!mb-8"
/>

<KaioNotification />

<Installation
  installCommand="yarn add @kaizen/components"
  importStatement='import { GlobalNotification } from "@kaizen/components"'
/>

## Overview

`<GlobalNotification>` is used to inform the user of system level messages.

<Canvas of={GlobalNotificationExamples.Playground} />

## API

<Controls of={GlobalNotificationExamples.Playground} />

### persistent

<Description of={GlobalNotificationExamples.Persistent} />
<Canvas of={GlobalNotificationExamples.Persistent} />
