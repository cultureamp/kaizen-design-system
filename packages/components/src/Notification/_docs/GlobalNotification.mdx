import { Canvas, Controls, Meta, Description} from "@storybook/blocks"
import { Installation, KaioNotification, ResourceLinks } from "~storybook/components"
import * as GlobalNotificationExamples from "./GlobalNotification.stories"

<Meta title="Components/Notifications/Global Notification" />

# GlobalNotification

<ResourceLinks
  sourceCode="https://github.com/cultureamp/kaizen-design-system/tree/main/packages/components/src/Notification/GlobalNotification"
  className="!mb-8"
/>

<KaioNotification />

<Installation
  installCommand="yarn add @kaizen/components"
  importStatement='import { GlobalNotification } from "@kaizen/components"'
/>

## Overview

`<GlobalNotification>` are used to inform the user of system level messages.

<Canvas of={GlobalNotificationExamples.Playground} />

## API

<Controls of={GlobalNotificationExamples.Playground} />

### persistent

<Description of={GlobalNotificationExamples.Persistent} />
<Canvas of={GlobalNotificationExamples.Persistent} />
