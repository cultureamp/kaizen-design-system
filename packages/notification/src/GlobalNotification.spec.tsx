import React from "react"
import { render } from "@testing-library/react"
import { GlobalNotification } from "./GlobalNotification"

describe("<GlobalNotification />", () => {
  it("renders the positive notification correctly", () => {
    const { container } = render(
      <GlobalNotification type="positive">
        This is my positive notification
      </GlobalNotification>
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the informative notification correctly", () => {
    const { container } = render(
      <GlobalNotification type="informative">
        This is my <em>informational</em> notification
      </GlobalNotification>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
