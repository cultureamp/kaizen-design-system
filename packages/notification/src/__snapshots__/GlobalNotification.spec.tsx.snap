// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalNotification /> renders the informative notification correctly 1`] = `
<div
  class="notification informative global hidden"
  data-automation-class="generic-notification informative global"
  style="margin-top: 0px;"
>
  <div
    class="icon"
  >
    <svg
      aria-hidden="true"
      class="icon inheritSize"
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </div>
  <div
    class="textContainer"
  >
    <div
      class="text"
    >
      This is my 
      <em>
        informational
      </em>
       notification
    </div>
  </div>
  <button
    class="cancel"
    data-testid="close-button"
    type="button"
  >
    <svg
      aria-label="close notification"
      class="icon"
      focusable="false"
      role="img"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </button>
</div>
`;

exports[`<GlobalNotification /> renders the positive notification correctly 1`] = `
<div
  class="notification positive global hidden"
  data-automation-class="generic-notification positive global"
  style="margin-top: 0px;"
>
  <div
    class="icon"
  >
    <svg
      aria-hidden="true"
      class="icon inheritSize"
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </div>
  <div
    class="textContainer"
  >
    <div
      class="text"
    >
      This is my positive notification
    </div>
  </div>
  <button
    class="cancel"
    data-testid="close-button"
    type="button"
  >
    <svg
      aria-label="close notification"
      class="icon"
      focusable="false"
      role="img"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </button>
</div>
`;
